VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.tsx

```diff
@@ -0,0 +1,273 @@
+            selectedFile: file
+        });
+    }
+
+    public uploadPhotoHandler = () => {
+        this.fileInputRef.current.click();
+    }
+
+    public saveQuizHandler = () => {
+
+        if(!this.allValid()){
+            return;
+        }
+        
+        const fd = new FormData();
+        fd.append('files', this.state.selectedFile);
+        fd.append('title', this.state.title.toUpperCase());
+
+        let url: string = '';
+        if(this.props.editMode){
+            url = 'http://localhost:60151/api/quizzes/' + this.props.quizId;
+        }else{
+            url = 'http://localhost:60151/api/quizzes';
+        }
+
+        this.httpService.postWithToken(url, fd)
+        .then((response: any) => {
+            const id: number = response.data.id;
+            const successButtonElement: any = document.querySelector(".success-btn");
+            successButtonElement.classList.add('hidden');
+            this.setState({
+                saved: true
+            })
+            this.props.save(id);
+        })
+        .catch((error: any) => console.log(error));
+    }
+
+    public render() {
+        return ( 
+            <div className = "start-quiz row">
+                <div id="title" className="col-md-5 col-xs-12" >
+                    <FormInput InputId="quizTitle" InputType="text" name="quizTitle" value={this.state.title} changed={this.changeTitleHandler} focus={this.focusTitleHandler} readonly={this.state.saved}/>
+                    <div className="error-message">{this.state.titleError}</div>
+                </div>
+                <div id="upload" className="col-md-5 col-xs-9">
+                    <input id="file-uploader" type="file" onChange={this.fileSelectHandler} ref={this.fileInputRef} />
+                    <span id="label-photo-txt"> { this.props.editMode ? 'change quiz photo' : this.state.selectedFile ? "edit quiz photo" : "upload quiz photo"}</span>
+                    <UploadButton click={this.uploadPhotoHandler} disabled={this.state.saved}/>
+                    {this.state.selectedFile ? <img id="preview-image" src={require("./../../../media/home.png")} alt="picture" ref={this.previewImageRef} /> : null}
+                    <div className="error-message">{this.state.imageError}</div>
+                </div>
+                <div className="success-btn col-md-2 col-xs-3">
+                    <SubmitButton click={this.saveQuizHandler}/>
+                </div>
+            </div>
+        )
+    }
+
+    public isTitleValid = (currentTitle: string): boolean => {
+
+        if(currentTitle === initialTitle || !currentTitle){
+            this.setState({
+                titleError: 'Please add a title'
+            });
+            this.addInvalidClassToTitle();
+            return false;
+        }
+        this.setState({
+            titleError: ''
+        });
+        this.removeInvalidClassFromTitle();
+        return true;
+    }
+
+    private allValid = (): boolean => {
```

Comment:
This method name doesn`t suggest me too many things. Please find a more meaningful name